### PR TITLE
Hardening/2193 entity not found when forwarding

### PR DESCRIPTION
**Pull request by kzangeli**
Two tests to ensure 404 Not Found is returned when a forward-updated entity doesn't exist

### CODE REVIEW

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+
+02. Modify E1/A1 in CP1 via CB, using PUT /v2/entities/E1/attrs/A1?type=T1, see error
+=====================================================================================
+HTTP/1.1 404 Not Found
+Content-Length: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
```
Comment (fgalan) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test]: Note this message hasn't payload (and it should). Probably this is due to the `"Fix: missing error payload for NotFound error in 'PUT /v2/entities/<eId>/attrs/<attrName>' and 'PUT /v2/entities/<eId>/attrs/<attrName>/value' operations"` has not reach this PR.

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+dbInit CB
+dbInit CP1
+brokerStart CB
+brokerStart CP1
+
+--SHELL--
+
+#
+# 01. Register E1/T1/A1 in CB, provApp: CP1
+# 02. Modify E1/A1 in CP1 via CB, using PUT /v2/entities/E1/attrs/A1?type=T
```
Comment (fgalan) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test]: "see error" not include in preamble comment.

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+
+02. Modify E1/A1 in CP1 via CB, using PUT /v2/entities/E1/attrs/A1?type=T1, see error
+=====================================================================================
+HTTP/1.1 404 Not Found
+Content-Length: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
```
Comment (fgalan) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test]: Note this message hasn't payload (and it should). Probably this is due to the "Fix: missing error payload for NotFound error in 'PUT /v2/entities/<eId>/attrs/<attrName>' and 'PUT /v2/entities/<eId>/attrs/<attrName>/value' operations" has not reach this PR.

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+dbInit CB
+dbInit CP1
+brokerStart CB
+brokerStart CP1
+
+--SHELL--
+
+#
+# 01. Register E1/T1/{} in CB, provApp: CP1
+# 02. Modify E1/A1 in CP1 via CB, using PUT /v2/entities/E1/attrs/A1?type=T
```
Comment (fgalan) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test]: "see error" not included in preamble comment.

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test]: Fixed in 7591449

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test (hunk 4 above)
Comment (kzangeli) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test]: Fixed in 7591449

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test (hunk 3 above)
Comment (fgalan) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_specific_attr.test]: Fixed in 1f12a84

File: test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test (hunk 1 above)
Comment (fgalan) [inline test/functionalTest/cases/2193_entity_not_found_when_forwarding/entity_not_found_when_forwarding_all_attrs.test]: Fixed 1f12a84

**Comment by fgalan** [comment]
LGTM

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**